graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -718,6 +718,21 @@ def _compile_ast_node_to_ir(schema, current_schema_type, ast, location, context)
     return basic_blocks
 
 
+def _validate_all_tags_are_used(context):
+    tag_names = set([tag_name for tag_name in context['tags']])
+    filter_args = set([])
+    for location, _ in context['metadata'].registered_locations:
+        filter_infos = context['metadata'].get_filter_infos(location)
+        for filter_info in filter_infos:
+            for arg in filter_info.args:
+                if _is_tag_argument(arg):
+                    filter_args.add(arg[1:])
```

Comment:
Add a comment explaining the structure of args. Think of this sort of comments as guide comments for future readers.

Reply:
We've added comments like that a lot, and it always felt a bit janky. Maybe it's time to have a helper method that gets the name of the argument without the prefix? Then we should be able to use it in `expressions.Variable` for validation as well, etc.

Reply:
temporarily added a comment. Is it better to create this method now and update other usages in another PR or leave this all for another PR?